src/Language/PureScript/CoreImp/ToJSON.hs

```diff
@@ -0,0 +1,150 @@
+            T.pack "args"       .= toJSON texts,
+            T.pack "body"        .= astToJSON ast ]
+astToJSON (Binary maybeSrcSpan binOp astLeft astRight) =
+    object [T.pack "sourceSpan" .= toJSON maybeSrcSpan,
+            T.pack "_Binary"    .= astToJSON astLeft,
+            T.pack "op"         .= show binOp,
+            T.pack "rhs"        .= astToJSON astRight ]
+astToJSON (ForIn maybeSrcSpan forname astInExpr astBody) =
+    object [T.pack "sourceSpan" .= toJSON maybeSrcSpan,
+            T.pack "_ForIn"     .= forname,
+            T.pack "in"        .= astToJSON astInExpr,
+            T.pack "body"        .= astToJSON astBody ]
+astToJSON (For maybeSrcSpan forname astCond astStep astBody) =
+    object [T.pack "sourceSpan" .= toJSON maybeSrcSpan,
+            T.pack "_ForIn"     .= forname,
+            T.pack "cond"        .= astToJSON astCond,
+            T.pack "step"        .= astToJSON astStep,
+            T.pack "body"        .= astToJSON astBody ]
+astToJSON (IfElse maybeSrcSpan astIf astThen astElse) =
+    object [T.pack "sourceSpan" .= toJSON maybeSrcSpan,
+            T.pack "_IfElse"    .= astToJSON astIf,
+            T.pack "then"       .= astToJSON astThen,
+            T.pack "else"       .= subAstToJSONMaybe astElse ]
+astToJSON (VariableIntroduction maybeSrcSpan text maybeAST) =
+    object [T.pack "sourceSpan"           .= toJSON maybeSrcSpan,
+            T.pack "_VariableIntroduction" .= toJSON text,
+            T.pack "rhs"                  .= subAstToJSONMaybe maybeAST ]
+astToJSON (ObjectLiteral maybeSrcSpan psStrASTs) =
+    object [T.pack "sourceSpan"    .= toJSON maybeSrcSpan,
+            T.pack "_ObjectLiteral" .= map psStrASTToJson psStrASTs ]
+astToJSON (ReturnNoResult maybeSrcSpan) =
+    object [T.pack "sourceSpan"      .= toJSON maybeSrcSpan,
+            T.pack "_ReturnNoResult" .= Null ]
+astToJSON (Return maybeSrcSpan ast) =
+    subASTSingleToJSON "_Return" maybeSrcSpan ast
+astToJSON (Throw maybeSrcSpan ast) =
+    subASTSingleToJSON "_Throw" maybeSrcSpan ast
+astToJSON (ArrayLiteral maybeSrcSpan asts) =
+    subASTsListToJSON "_ArrayLiteral" maybeSrcSpan asts
+astToJSON (Assignment maybeSrcSpan astLeft astRight) =
+    subASTsLeftRightToJSON "_Assignment" maybeSrcSpan astLeft astRight
+astToJSON (Indexer maybeSrcSpan astInner astOuter) =
+    subASTsLeftRightToJSON "_Indexer" maybeSrcSpan astOuter astInner
+astToJSON (InstanceOf maybeSrcSpan astLeft astRight) =
+    subASTsLeftRightToJSON "_InstanceOf" maybeSrcSpan astLeft astRight
+
+psStrASTToJson :: (PSString, AST) -> Value
+psStrASTToJson (psStr, ast) =
+    object [T.pack (decodeStringWithReplacement psStr) .= astToJSON ast]
+
+subAstToJSONMaybe :: Maybe AST -> Value
+subAstToJSONMaybe (Just ast) =
+    astToJSON ast
+subAstToJSONMaybe Nothing =
+    Null
+
+subASTSingleToJSON :: String -> (Maybe SourceSpan) -> AST -> Value
+subASTSingleToJSON name maybeSrcSpan ast =
+    object [T.pack "sourceSpan" .= toJSON maybeSrcSpan,
```

Comment:
Does this hit the same issue as the CoreFn change, where it stores the file path in every node?

Reply:
Well with `--source-maps` enabled. Is it an issue to begin with? We're emitting *sizable* to-be-machine-processed json files already.. I can certainly ditch the file path if you guys prefer, personally am hoping I won't need the sourcemappings for my ps2go transpiler anyway